core/examples/restore_checkpoint_delta.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright 2019 Conflux Foundation. All rights reserved.
+// Conflux is free software and distributed under GNU General Public License.
+// See http://www.gnu.org/licenses/
+
+use cfx_types::{Address, H256};
+use cfxcore::{
+    db::NUM_COLUMNS,
+    statedb::StateDb,
+    storage::{
+        state::StateTrait,
+        state_manager::{
+            StateManager, StateManagerTrait, StorageConfiguration,
+        },
+        SnapshotAndEpochIdRef,
+    },
+    sync::{
+        delta::{Chunk, ChunkReader, StateDumper},
+        restore::Restorer,
+        Error,
+    },
+};
+use primitives::{Account, MerkleHash};
+use rlp::Rlp;
+use std::{
+    cmp::min,
+    fs::{create_dir_all, remove_dir_all},
+    path::{Path, PathBuf},
+    str::FromStr,
+    sync::Arc,
+    time::{Duration, Instant},
+};
+
+// cargo run --release -p cfxcore --example restore_checkpoint_delta
+fn main() -> Result<(), Error> {
+    let test_dir = PathBuf::from("E:\\zzzzzz");
+    if test_dir.exists() {
+        remove_dir_all(&test_dir)?;
+    }
+
+    // setup node 1
+    println!("====================================================");
+    println!("Setup node 1 ...");
+    let sm1 = new_state_manager(test_dir.join("db1").to_str().unwrap())?;
+    let (genesis_hash, genesis_root) = initialize_genesis(&sm1)?;
+    let (checkpoint, checkpoint_root) =
+        prepare_checkpoint(&sm1, &genesis_hash, 5_000_000)?;
```

Comment:
You can prepare a large checkpoint, but it's not possible to restore a big DeltaMPT due to small dirty_size config parameter. Make sure you adjust the dirty_size parameter and change the number accordingly.